### PR TITLE
Variorum Connector

### PR DESCRIPTION
**Pull request by zfrye-llnl**
Added a develop version of a simple Kokkos and Variorum connector.

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
The code looks good, generally. I'll take it for a run later in the week and review after I have some experience using it

File: profiling/variorum-connector/variorum-connector.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+time_t start_time;
+
+
+extern "C" void kokkosp_init_library(const int loadSeq,
+	const uint64_t interfaceVer, const uint32_t devInfoCount, void* deviceInfo) {
+    
+    int ret;
+    //Simple timer code to keep track of the general amount of time the application ran for.
+    time(&start_time);
+    std::cout << "Start Time: " << start_time << std::endl;
```
Comment (dalg24) [inline profiling/variorum-connector/variorum-connector.cpp:95]: I wouldn't flush the stream here.
```suggestion
    std::cout << "Start Time: " << start_time << '\n';
```
or use `printf`

File: profiling/variorum-connector/variorum-connector.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    //print variorum power when kokkos is finalized is called
+	ret = variorum_print_power();
+    if (ret != 0) {
+        printf("Print power failed!\n");
+    }
+    time_t total_time;
+    time_t end_time;
+    time(&end_time);
+    std::cout << "End Time: " << end_time << "\nStart Time: " << start_time << std::endl;
+    total_time = end_time - start_time;
```
Comment (dalg24) [inline profiling/variorum-connector/variorum-connector.cpp:120]:
```suggestion
    time_t end_time;
    time(&end_time);
    std::cout << "End Time: " << end_time << "\nStart Time: " << start_time << std::endl;
    time_t total_time = end_time - start_time;
```

File: profiling/variorum-connector/variorum-connector.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+}
+
+extern "C" void kokkosp_begin_parallel_for(const char* name, const uint32_t devID, uint64_t* kID) {
+
+    int ret;
+    //print variorum power whenever parallelfor is called
+	ret = variorum_print_power();
+    if (ret != 0) {
+        printf("Print power failed!\n");
+    }
```
Comment (dalg24) [inline profiling/variorum-connector/variorum-connector.cpp:132]: Consider defining a function to avoid code duplication.

**Comment by zfrye-llnl** [comment]
I've made the changes recommended in these comments. Feel free to let me know if there are any other changes that should be done.

**Comment by tpatki** [comment]
Couple of things to work on in the next version after this PR is merged: 

(1) Use the JSON-based API instead of the print-power API in variorum, that way we can write to a file as opposed to stdout. 
(2)  Create a per node/per rank output file with power data (maybe look at how PAPI does this?)

@zfrye-llnl @DavidPoliakoff

**Comment by DavidPoliakoff** [comment]
Sorry, hadn't realized I hadn't hit the merge button yet. Feel free to update if you have any changes

**Pull request merged**